aioxmpp/ibr/service.py

```diff
@@ -0,0 +1,253 @@
+        yield from client.send(iq)
+
+    @asyncio.coroutine
+    def cancel_registration(self, client):
+        """
+        Cancels the currents client's account with the server.
+
+        :param client: Client who wants to cancel.
+        :type client: :class:`aioxmpp.Client`
+
+        :return:`None`
+
+        Even if the cancelation is succesful, this method will raise an
+        exception due to he account no longer exists for the server, so the
+        client will fail.
+        To continue with the execution, this method should be surrounded by a
+        try/except statement.
+        """
+        iq = aioxmpp.IQ(
+            to=client.local_jid.bare().replace(localpart=None),
+            type_=aioxmpp.IQType.SET,
+            payload=xso.Query()
+        )
+
+        iq.payload.remove = True
+        yield from client.send(iq)
+
+    @staticmethod
+    @asyncio.coroutine
+    def connect_and_send(iq, loop, metadata, peer_jid, timeout):
+        options = list((
+            yield from aioxmpp.discover_connectors(
+                peer_jid.domain, loop=loop
+            )
+        ))
+        for host, port, conn in options:
+            transport, xmlstream, features = yield from conn.connect(
+                loop,
+                metadata,
+                peer_jid.domain,
+                host,
+                port,
+                timeout,
+                base_logger=logger
+            )
+            reply = yield from aioxmpp.send_and_wait_for(
+                xmlstream, [iq], [aioxmpp.IQ]
+            )
+        return reply
+
+    @asyncio.coroutine
+    def get_registration_fields(self,
+                                peer_jid,
+                                timeout=60,
+                                no_verify=True
+                                ):
+        """
+        A query is sent to the server to obtain the fields that need to be
+        filled to register with the server.
+
+        :param peer_jid: Server where he query will be sent.
+        :type peer_jid: :class:`aioxmpp.JID`
+
+        :param timeout: Maximum time in seconds to wait for an IQ response, or
+                        :data:`None` to disable the timeout.
+        :type timeout: :class:`~numbers.Real` or :data:`None`
+
+        :param no_verify: Specifies whether the server's TLS is verified.
+        :type no_verify: :class:`aioxmpp.JID`
+
+        :return: :attr:`list`
+        """
+        iq = aioxmpp.IQ(
+            to=peer_jid.bare().replace(localpart=None),
+            type_=aioxmpp.IQType.GET,
+            payload=xso.Query()
+        )
+        iq.autoset_id()
+
+        loop = asyncio.get_event_loop()
+        metadata = aioxmpp.make_security_layer('', no_verify=no_verify)
+
+        try:
+            reply = yield from self.connect_and_send(iq, loop, metadata, peer_jid, timeout)
+
+            return [a for a in dir(reply.payload)
+                    if isinstance(getattr(reply.payload, a), str) and
```

Comment:
This check can be made more readable by checking against ``is not None`` instead of ``isinstance(…, str)``. The type can only be str or NoneType, and it not being None is what you’re really interested in (because that means it has not been defaulted).